pkg/kubelet/volume/gce_pd/gce_pd.go

```diff
@@ -194,9 +194,20 @@ func (pd *gcePersistentDisk) SetUp() error {
 	globalPDPath := makeGlobalPDName(pd.plugin.host, pd.pdName, pd.readOnly)
 	err = pd.mounter.Mount(globalPDPath, pd.GetPath(), "", mount.FlagBind|flags, "")
 	if err != nil {
-		os.RemoveAll(pd.GetPath())
-		// TODO: we should really eject the attach/detach out into its own control loop.
-		detachDiskLogError(pd)
+		mountpoint, mntErr := isMountPoint(pd.GetPath())
+		if mntErr != nil {
+			glog.Errorf("isMountpoint check failed: %v", mntErr)
+			return err
+		}
+		if !mountpoint {
```

Comment:
I'd rather see this structured a bit more obviously:

```
... = isMountPoint()
if mntErr {  log it & return }
if mountpoint {
    call unmount
    ... = isMountPoint() // again
    if mountpoint {
        // Comment saying this is very odd, and we'll try again next sync loop
        log(this is bad, we'll try again next sync loop)
        return
    }
    os.Remove()
    detach
```

Specifically the comment  that we will try again is important to comprehension.